File: gencode/python/udmi/schema/building_translation.py
"""Generated class for building_translation.json"""


class BuildingTranslation:
  """Generated schema class"""

  def __init__(self):
    self.present_value2 = None
    self.units = None
    self.ref = None
    self.states = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = BuildingTranslation()
    result.present_value2 = source.get('present_value2')
    result.units = source.get('units')
    result.ref = source.get('ref')
    result.states = source.get('states')
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = BuildingTranslation.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.present_value2:
      result['present_value2'] = self.present_value2 # 5
    if self.units:
      result['units'] = self.units # 5
    if self.ref:
      result['ref'] = self.ref # 5
    if self.states:
      result['states'] = self.states # 5
    return result

File: gencode/java/udmi/schema/BuildingTranslation.java

package udmi.schema;

import javax.annotation.processing.Generated;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Building Translation
 * <p>
 * [Discovery result](../docs/specs/discovery.md) with implicit enumeration
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "present_value2",
    "units",
    "ref",
    "states"
})
@Generated("jsonschema2pojo")
public class BuildingTranslation {

    /**
     * dotted path to present_value field
     * 
     */
    @JsonProperty("present_value2")
    @JsonPropertyDescription("dotted path to present_value field")
    public String present_value2;
    @JsonProperty("units")
    public Object units;
    @JsonProperty("ref")
    public String ref;
    @JsonProperty("states")
    public Object states;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.ref == null)? 0 :this.ref.hashCode()));
        result = ((result* 31)+((this.units == null)? 0 :this.units.hashCode()));
        result = ((result* 31)+((this.present_value2 == null)? 0 :this.present_value2 .hashCode()));
        result = ((result* 31)+((this.states == null)? 0 :this.states.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof BuildingTranslation) == false) {
            return false;
        }
        BuildingTranslation rhs = ((BuildingTranslation) other);
        return (((((this.ref == rhs.ref)||((this.ref!= null)&&this.ref.equals(rhs.ref)))&&((this.units == rhs.units)||((this.units!= null)&&this.units.equals(rhs.units))))&&((this.present_value2 == rhs.present_value2)||((this.present_value2 != null)&&this.present_value2 .equals(rhs.present_value2))))&&((this.states == rhs.states)||((this.states!= null)&&this.states.equals(rhs.states))));
    }

}
